pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -174,6 +179,18 @@ func (mixerplugin) OnInboundListener(in *plugin.InputParams, mutable *plugin.Mut
 	return fmt.Errorf("unknown listener type %v in mixer.OnOutboundListener", in.ListenerProtocol)
 }
 
+// OnVirtualListener implements the Plugin interface method.
+func (mixerplugin) OnVirtualListener(in *plugin.InputParams, mutable *plugin.MutableObjects) error {
+	if in.ListenerProtocol == plugin.ListenerProtocolTCP {
+		attrs := createOutboundListenerAttributes(in)
+		tcpFilter := buildOutboundTCPFilter(in.Env.Mesh, attrs, in.Node, in.Service, in.Push)
```

Comment:
I'm confused -- is this virtual listener used only for blackhole/passthrough routes? It used to be that the virtual listener would handle _all_ outbound traffic and then route to concrete listeners using use_original_port or something like that.
If it only handle blackhole/passthrough traffic, can that be clear in the plugin interface API?

Reply:
Virtual Listeners currently act as the receiver of ip tables redirect, then use SO_ORIG_DST for handing it off to other listeners based on original destination and if not fall back to BlackHole or Passthrough based on global configuration.

I feel like the name is appropriate but open to suggestions.

Reply:
Can you confirm that the filter chain in the virtual listener is not invoked when a concrete listener matching SO_ORIG_DST is found?

Reply:
Fine with the name and the change as long as you confirm that we don't double-call mixerfilter here.

Reply:
I have verified it manually but I was hoping/relying on automated tests to fail if extraneous metrics were emitted.
I can do more extensive testing and cover any test case you had in mind.

Reply:
Manual check is OK here. I am pretty sure this code is subject to change really-soon-now due to filter chain refactor, so there's no need to invest in automation for now. Any of our e2e should have caught double metric reporting hopefully.